guides/v15-mdc-migration.md

```diff
@@ -0,0 +1,571 @@
+  `<input matInput>` without the form field if the page loaded form field styles elsewhere.
+
+* The MDC-based MatInput hides the native calendar picker indicator associated with
+  `<input matInput type="date">`, if you want this indicator to appear for your inputs, use the
+  following CSS:
+
+    ```scss
+    .mat-mdc-input-element::-webkit-calendar-picker-indicator {
+      display: block;
+    }
+    ```
+
+### List
+
+* The API has been reworked to support text wrapping and better integration with the Material Design
+  specification.
+
+* Previously, list items were commonly created using multiple `span` elements with the `matLine`
+  directive applied. Each `span` resulting in a line, and the first one becoming the primary line.
+  With the new API, the `matLine` directive has been split into two more granular and obvious
+  directives:
+  * `matListItemTitle`
+  * `matListItemLine`
+
+* Every list item is supposed to have a title for accessibility. If no explicit title and lines are
+  provided, the direct content of the `<mat-list-item>` will become the title. You cannot have
+  multiple line directives without a title.
+
+  ```html
+  <mat-list-item>This is the title</mat-list-item>
+  ```
+  
+* Text outside of a `matListItemLine` (so-called "unscoped content") will result in an additional
+  line being acquired (as if the content was put into a line).
+
+  ```html
+  <mat-list-item>
+    <span matListItemTitle>Title</span>
+    Second line
+  </mat-list-item>
+  ```
+  
+* The amount of lines is automatically inferred. e.g. in the snippet above the list item will
+  acquire space for two lines. With the new API you can now set an explicit number of lines on the
+  `<mat-list-item>` to activate wrapping.
+
+  ```html
+  <mat-list-item lines="3">
+    <span matListItemTitle>Title</span>
+    This will text will wrap into the third line. Space for three lines is acquired by the list
+    item.
+  </mat-list-item>
+  ```
+
+  Note that text inside a `matListItemTitle` or `matListItemLine` will never wrap. Only unscoped
+  content will wrap/take up the remaining space based on explicit number of lines provided.
+
+* Aside from the differences in how lines are composed, some other directives have been renamed to
+  use more explicit naming:
+  * `matListIcon` is now `matListItemIcon`
+  * `matListAvatar` is now `matListItemAvatar`
+
+* Lastly, also a new directive (`matListItemMeta`) is available to put content into the meta section
+  of a list item (usually the end of the list item). Previously unscoped content in a list item was 
+  put into the meta section.
+
+* Recommended migration steps for common use of a list item:
+  1. Change the first `matLine` to `matListItemTitle`
+  2. Change all other `matLine`'s to `matListItemLine`
+  3. Change all `matListIcon` to `matListItemIcon`
+  4. Change all `matListAvatar` to `matListItemAvatar`
+  5. Wrap all unscoped content (content outside a `matLine`) in a `matListItemMeta` container.
```

Comment:
cc @wagnermaciel how much of this do we do automatically?

Reply:
FYI: In the internal migration tool- I implemented an initial AST-based automation of this. It's hard to get this fully right but it leaves TODOs in place and tries to help at least a bit.

Reply:
We probably want to port that logic to the external tool if we haven't already @amysorto